### ISSUE TITLE
linter: `jsx-a11y/prefer-tag-over-role` is enabled by default

### ISSUE THREAD

**Issue by elusiveunit** [oxc-project/oxc#23491]
### What version of Oxlint are you using?

1.70.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```typescript
export default defineConfig({
	plugins: ['jsx-a11y'],
	rules: {},
});
```

### What happened?

Testing.tsx
```
export default function Testing() {
	return <span role="paragraph">Hello</span>;
}
```
Output from `npx oxlint`:
```
  ⚠ jsx-a11y(prefer-tag-over-role): Prefer `p` over `role` attribute `paragraph`.
   ╭─[Testing.tsx:2:15]
 1 │ export default function Testing() {
 2 │     return <span role="paragraph">Hello</span>;
   ·                  ────────────────
 3 │ }
   ╰────
  help: Replace HTML elements with `role` attribute `paragraph` to corresponding semantic HTML tag `p`.
```
This could be considered a docs issue for the website rather than a linter bug, but since [upstream](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/prefer-tag-over-role.md) ([rule list](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y#supported-rules)) doesn't have it enabled by default I'm guessing oxlint shouldn't either?

**Comment by camc314** [comment]
Hi, thanks for making this issue.

Our rule categories odon'es necessarily align with the upstream rule.

In this case, we have decided that this rule fits into the "correctness" category, so is on by default whe the `jsx-a11y` category is enabled.

It has been this way since the rule was introduced (1+ years ago).

I think correctness fits is ok for this rule, as the rule is pushing users towards using the correct semantic element type rather than an arbitary element + using `role`.

That said, I'm closing this as not planned - if you want to disable this rule add:
```
{
    "plugins": ["jsx-a11y"],
    "rules": { 
        "jsx-a11y/refer-tag-over-role": "off"
    }
}
```

Thanks
